src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_primitive_fusing.cpp

```diff
@@ -186,17 +186,24 @@ void prepare_primitive_fusing::fuse_swiglu(program &p) {
             if (!node->get_dependency(0).is_type<fully_connected>())
                 continue;
             auto swiglu_prim = node->get_kernel_impl_params()->typed_desc<swiglu>();
-            auto& fc_node = node->get_dependency(0);
+            auto& fc_node = node->get_dependency(0).as<fully_connected>();
             if (node->get_dependencies().size() > 1)
                 continue;
-            if (!node->get_dependency(0).get_fused_primitives().empty())
+            if (!fc_node.get_fused_primitives().empty())
                 continue;
             auto in_dt = fc_node.get_input_layout(0).data_type;
             if (in_dt != data_types::f16)
                 continue;
             auto wt_dt = fc_node.get_input_layout(1).data_type;
             if (!data_type_traits::is_i4_u4(wt_dt))
                 continue;
+            // TODO: For per-channel quantized models(# of decompression scale groups = 1), 2FCs+SwiGLU fusion is disabled due to accuracy issue
+            bool has_scale = !fc_node.get_primitive()->decompression_scale.empty();
+            size_t scale_idx = fc_node.get_primitive()->bias.empty() ? 2 : 3;
+            if (has_scale &&
+                fc_node.get_input_layout(scale_idx).is_static() &&
+                fc_node.get_input_layout(scale_idx).feature() == 1)
```

Comment:
I'd like to suggest to use the condition as [scale shape's entire # element == fc node's weight's batch size] because we cannot ensure that the scale input's feature is always corresponding to the K dimension.

Reply:
applied